yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-row-counts.sqlplus

```diff
@@ -14,21 +14,89 @@ WHENEVER OSERROR EXIT FAILURE
 -- Select the data into the CSV file using inline views
 SPOOL table-row-counts.csv
 
-SELECT
-    s.owner AS schema_name,
-    CASE
-        WHEN s.segment_type = 'TABLE' THEN s.segment_name
-        WHEN s.segment_type = 'TABLE PARTITION' THEN s.segment_name || '_' || s.partition_name
-        WHEN s.segment_type = 'TABLE SUBPARTITION' THEN s.segment_name || '_' || sp.partition_name || '_' || s.partition_name
-    END AS table_name,
-    0 as row_count
-FROM
-    dba_segments s
-LEFT JOIN
-    dba_tab_subpartitions sp ON s.owner = sp.table_owner AND s.segment_name = sp.table_name AND s.partition_name = sp.subpartition_name
-WHERE
-    s.segment_type IN ('TABLE', 'TABLE PARTITION', 'TABLE SUBPARTITION')
-    AND s.owner = '&1';
+-- Inline view to get the base tables, partitions, and subpartitions with correct naming
```

Comment:
@sanyamsinghal check this. I think this should do it...
```
SELECT DISTINCT 
    t.owner AS schema_name, 
    t.table_name,
    CASE 
      WHEN sp.subpartition_name IS NOT NULL THEN sp.table_name || '_' || sp.partition_name || '_' || sp.subpartition_name 
      WHEN  tp.partition_name IS NOT NULL THEN tp.table_name || '_' || tp.partition_name
      ELSE t.table_name 
    END AS table_name_generated,
    0 as row_count
  FROM
    dba_tables t
  LEFT JOIN 
    dba_tab_partitions tp ON t.owner = tp.table_owner AND t.table_name = tp.table_name
  LEFT JOIN
    dba_tab_subpartitions sp ON t.owner = sp.table_owner AND t.table_name = sp.table_name 
  WHERE 
    t.owner = 'SDAB';
```

Reply:
updated fyi

Reply:
```
SELECT  
    t.owner AS schema_name, 
    t.table_name,
    CASE 
      WHEN sp.subpartition_name IS NOT NULL THEN sp.table_name || '_' || sp.partition_name || '_' || sp.subpartition_name 
      WHEN  tp.partition_name IS NOT NULL THEN tp.table_name || '_' || tp.partition_name
      ELSE t.table_name 
    END AS table_name_generated,
    0 as row_count
  FROM
    dba_tables t
  LEFT JOIN 
    dba_tab_partitions tp ON t.owner = tp.table_owner AND t.table_name = tp.table_name
  LEFT JOIN
    dba_tab_subpartitions sp ON tp.table_owner = sp.table_owner AND tp.partition_name = sp.partition_name AND t.table_name = sp.table_name
  WHERE 
    t.owner = 'ASSESS_MIGRATION';
```

Reply:
done